File: 907-koko-eating-bananas/koko-eating-bananas.py
import math
class Solution:
    def helper(self,piles,mid):
        ans=0
        for i in range(len(piles)):
            ans=ans+math.ceil(piles[i]/mid)
        return ans
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        lo=1
        hi=max(piles)
        ans=-1
        while lo<=hi:
            mid=lo+(hi-lo)//2
            if self.helper(piles,mid)>h:
                lo=mid+1
            else:
                ans=mid
                hi=mid-1
        return ans

        

File: 907-koko-eating-bananas/koko-eating-bananas.java
class Solution {

    public boolean helper(int []piles, int mid, int h){
        long countHours=0;
        int length=piles.length;
        for(int i=0;i<length;i++){
            countHours = countHours + (int) Math.ceil((double) piles[i]/ (double) mid);
        }
        if(countHours<=h){
            return true;
        }
        return false;
    }
    public int minEatingSpeed(int[] piles, int h) {
        int low=1;
        int high=Arrays.stream(piles).max().getAsInt();
        int ans=-1;
        while(low<=high){
            int mid=(low+high)/2;
            boolean flag=helper(piles,mid,h);
            if(flag==true){
                ans=mid;
                high=mid-1;
            }else{
                low=mid+1;
            }
        }
        return ans;
    }
}